doc/source/how-to-configure-clients.rst

```diff
@@ -29,115 +28,151 @@ them back to Python dictionaries.
-    value types (and converting them back on the client-side).
+    ``List``, ``Map``) as values in configuration dictionaries. To send collections,
+    convert them to a supported type (e.g., JSON string) and decode on the client side.
 
-    One can, for example, convert a list of floating-point numbers to a JSON string,
-    then send the JSON string using the configuration dictionary, and then convert the
-    JSON string back to a list of floating-point numbers on the client.
+    Example:
 
-Configuration through built-in strategies
------------------------------------------
+    .. code-block:: python
 
-The easiest way to send configuration values to clients is to use a built-in strategy
-like ``FedAvg``. Built-in strategies support so-called configuration functions. A
-configuration function is a function that the built-in strategy calls to get the
-configuration dictionary for the current round. It then forwards the configuration
-dictionary to all the clients selected during that round.
+        import json
 
-Let's start with a simple example. Imagine we want to send (a) the batch size that the
-client should use, (b) the current global round of federated learning, and (c) the
-number of epochs to train on the client-side. Our configuration function could look like
-this:
+        # On the server
+        config_dict = {"data_splits": json.dumps([0.8, 0.1, 0.1])}
+
+        # On the client
+        data_splits = json.loads(config["data_splits"])
+
+Using Built-in Strategies for Configuration
+-------------------------------------------
+
+Flower supports configuration functions to dynamically adjust parameters sent to
+clients. Built-in strategies like ``FedAvg`` allow for setting configuration values for
+each round via a function.
+
+Example: Sending Training Configurations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Imagine we want to send (a) the batch size, (b) the current global round, and (c) the
+number of local epochs. Our configuration function could look like this:
 
 .. code-block:: python
 
     def fit_config(server_round: int):
-        """Return training configuration dict for each round."""
-        config = {
+        """Generate training configuration for each round."""
+        return {
             "batch_size": 32,
             "current_round": server_round,
             "local_epochs": 2,
         }
-        return config
 
-To make the built-in strategies use this function, we can pass it to ``FedAvg`` during
-initialization using the parameter ``on_fit_config_fn``:
+To use this function with a built-in strategy like ``FedAvg``, pass it during
+initialization:
 
 .. code-block:: python
 
-    strategy = FedAvg(
-        ...,  # Other FedAvg parameters
-        on_fit_config_fn=fit_config,  # The fit_config function we defined earlier
-    )
+    strategy = FedAvg(on_fit_config_fn=fit_config)  # Assign the configuration function
```

Comment:
I think it's better to remove this line and present it directly as part of `server_fn`

Reply:
done